fullstack-cert/typescript-projects/digital-pet-game/src/App.tsx

```diff
@@ -0,0 +1,287 @@
+import { useState, useEffect, useRef } from "react";
+import "./App.css";
+import * as catFacts from "./assets/cat-facts.json";
+
+interface StatBarProps {
+  label: string;
+  value: number;
+  icon: string;
+}
+
+function getStatColorClass(value: number): string {
+  if (value >= 70) return "high";
+  if (value >= 40) return "medium";
+  return "low";
+}
+
+function StatBar({ label, value, icon }: StatBarProps) {
+  const colorClass = getStatColorClass(value);
+
+  return (
+    <div className="stat-bar">
+      <div className="stat-header">
+        <div className="stat-label">
+          <span className="stat-icon">{icon}</span>
+          <span className="stat-name">{label}</span>
+        </div>
+        <span className="stat-value">{Math.round(value)}%</span>
+      </div>
+
+      <div className="stat-progress">
+        <div
+          className={`stat-fill ${colorClass}`}
+          style={{ width: `${value}%` }}
+        />
+      </div>
+    </div>
+  );
+}
+
+enum PetAction {
+  NONE,
+  EAT,
+  PLAY,
+  SLEEP,
+}
+
+interface Pet {
+  name: string;
+  happiness: number;
+  hunger: number;
+  energy: number;
+  species: string;
+  action: PetAction;
+}
+
+enum PetMood {
+  Happy,
+  Excited,
+  Content,
+  Sad,
+  Tired,
+  Sick,
+  Hungry,
+}
+
+export function calculatePetMood(pt: Pet): PetMood {
+  const { hunger, happiness, energy /*health*/ } = pt;
+
+  //if (health < 30) return PetMood.Sick;
+  if (hunger > 70) return PetMood.Hungry;
+  if (energy < 30) return PetMood.Tired;
+  if (happiness < 30) return PetMood.Sad;
+  if (happiness > 80 && energy > 70) return PetMood.Excited;
+  if (happiness > 60) return PetMood.Happy;
+
+  return PetMood.Content;
+}
+
+const moodEmojiMap: Record<PetMood, string> = {
+  [PetMood.Happy]: "😺", // grinning cat
+  [PetMood.Excited]: "😻", // heart-eyes cat
+  [PetMood.Content]: "😸", // smiling cat
+  [PetMood.Sad]: "😿", // crying cat
+  [PetMood.Tired]: "😽", // kissing cat (closest to sleepy)
+  [PetMood.Sick]: "🙀", // weary cat (looks unwell)
+  [PetMood.Hungry]: "😹", // joy cat (closest to hungry/funny)
+};
+
+const saveKey = "pet";
+
+export function App() {
+  //TODO: The pet has a lot of functionality tied to it that's currently
```

Comment:
Since this is a lab, we can take either approach - our solution code here will ONLY be used for automated testing. Campers will implement their own approach to fulfill the user stories.